azure-iot-provisioning-devicessdk/azure/iot/provisioning/devicessdk/security/sk_security_provider.py

```diff
@@ -0,0 +1,64 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains a provider that is responsible for providing shared access tokens that will eventually establish
+ the authenticity of devices to Device Provisioning Service.
+"""
+from azure.iot.common.sastoken import SasToken
+
+
+class SymmetricKeySecurityProvider(object):
+    """
+    A provider that is responsible for providing shared access tokens that will eventually establish
+    the authenticity of devices to Device Provisioning Service.
+    """
+
+    def __init__(self, registration_id, symmetric_key, id_scope):
+        """
+        Initialize the symmetric key security client.
```

Comment:
>client [](start = 46, length = 6)

provider?

Reply:
let's make it client everywhere as other SDKs use that - then we can discuss what the better name is.

---
In reply to: [265317526](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r265317526) [](ancestors = 265317526)